src/core/profiling/endoscope_backend.c

```diff
@@ -0,0 +1,587 @@
+#include "src/core/profiling/endoscope_backend.h"
```

Comment:
Naming style is all over the place in this file. Please make it consistent with the rest of the core codebase.

function_names_are_written_like_this.